File: android/app/src/main/java/com/example/demointegratern/TestConnectNativeModule.java
package com.example.demointegratern;

import android.content.Intent;
import android.util.Log;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;

public class TestConnectNativeModule extends ReactContextBaseJavaModule {
    private static ReactApplicationContext reactContext;

    TestConnectNativeModule(ReactApplicationContext context) {
        super(context);
        reactContext = context;
    }

    @Override
    public String getName() {
        return "TestConnectNative";
    }

    @ReactMethod
    public void sendMessageToNative(String rnMessage) {
        Log.d("This log is from java", rnMessage);
    }

    @ReactMethod
    public void sendCallbackToNative(Callback rnCallback) {
        rnCallback.invoke("A greeting from java");
    }

    @ReactMethod
    public void finishActivity() {
        if (getCurrentActivity() != null) {
            getCurrentActivity().finish();
        }
    }

    @ReactMethod
    public void goToSecondActivity() {
        if (getCurrentActivity() != null) {
            Intent intent = new Intent(getCurrentActivity(), SecondActivity.class);
            getCurrentActivity().startActivity(intent);
        }
    }
}

File: android/app/src/main/java/com/example/demointegratern/TestConnectNativeModule.kt
package com.example.demointegratern

import android.content.Intent
import android.util.Log
import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class TestConnectNativeModule internal constructor(private val reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName(): String {
        return "TestConnectNative"
    }

    @ReactMethod
    fun sendMessageToNative(rnMessage: String?) {
        Log.d("This log is from java", rnMessage)
    }

    @ReactMethod
    fun sendCallbackToNative(rnCallback: Callback) {
        rnCallback.invoke("A greeting from java")
    }

    @ReactMethod
    fun finishActivity() {
        if (currentActivity != null) {
            currentActivity!!.finish()
        }
    }

    @ReactMethod
    fun goToSecondActivity() {
        if (currentActivity != null) {
            val intent = Intent(currentActivity, SecondActivity::class.java)
            currentActivity!!.startActivity(intent)
        }
    }

    companion object

}